apps/laboratory/src/components/Connection/ConnectionAccount.tsx

```diff
@@ -0,0 +1,126 @@
+import { Badge, Button, Flex, HStack, Text } from '@chakra-ui/react'
+
+import type { ChainNamespace } from '@reown/appkit-common'
+import { type Connection, useAppKitAccount } from '@reown/appkit/react'
+
+interface ConnectionAccountProps {
+  connection: Connection
+  namespace: ChainNamespace
+  account: {
+    address: string
+  }
+  activeConnection?: Connection
+  isLoading: boolean
+  active: boolean
+  onDelete: (address: string, connectorId: string) => void
+  onDisconnect: (connection: Connection) => void
+  onSwitch: (address: string, connection: Connection) => void
+  lastSelectedConnection:
+    | {
+        connectorId: string
+        address: string
+      }
+    | undefined
+  setLastSelectedConnection: (lastSelectedConnection: {
+    connectorId: string
+    address: string
+  }) => void
+}
+
+export function ConnectionAccount({
+  connection,
+  namespace,
+  account,
+  active,
+  activeConnection,
+  isLoading,
+  lastSelectedConnection,
+  setLastSelectedConnection,
+  onDelete,
+  onDisconnect,
+  onSwitch
+}: ConnectionAccountProps) {
+  const { address } = useAppKitAccount({ namespace })
+
+  const isCurrentActiveAccount = account.address === address
+  const isCurrentActiveConnection = connection.connectorId === activeConnection?.connectorId
+
+  const isSwitching =
+    isLoading &&
+    lastSelectedConnection?.connectorId === connection.connectorId &&
+    lastSelectedConnection?.address === account.address
+
+  const shortenedAddress = `${account.address.slice(0, 6)}...${account.address.slice(-4)}`
```

Comment:
If somehow `account.address` is undefined this `slice` will throw error - is `account.address` has value on any case?